File: kaltura/api_v3/lib/types/KalturaUiConf.php
<?php
/**
 * @package api
 * @subpackage objects
 */
class KalturaUiConf extends KalturaObject 
{
	/**
	 * @var string
	 * @readonly
	 * @filter eq,not,in,notIn,gte
	 */
	public $id;
	
	/**
	 * Name of the uiConf, this is not a primary key
	 * @var string
	 */
	public $name;
	
	/**
	 * 
	 * @var string
	 */
	public $description;
	
	/**
	 * 
	 * @var int
	 * @readonly
	 */
	public $partnerId;	
	

	/**
	 * @var KalturaUiConfObjType $objType
	 */
	public $objType;

	/**
	 * @var string
	 * @readonly
	 */
	public $objTypeAsString;
	
	/**
	 *
	 * @var int
	 */
	public $width;
	
	/**
	 * @var int
	 */
	public $height;
	
	/**
	 * @var string
	 */
	public $htmlParams;
	
	/**
	 * @var string
	 */
	public $swfUrl;
	
	/**
	 * @var string
	 * @readonly
	 */
	public $confFilePath;
	
	/**
	 * @var string
	 */
	public $confFile;

	/**
	 * @var string
	 */
	public $confFileFeatures;
	
	/**
	 * @var string
	 */
	public $confVars;
	
	
	/**
	 * @var bool
	 */
	public $useCdn;
	
	/**
	 * @var string
	 */
	public $tags;
	
	
	/**
	 * @var string
	 */
	public $swfUrlVersion;
	
	
	/**
	 * Entry creation date as Unix timestamp (In seconds)
	 * @var int
	 * @readonly
	 */
	public $createdAt;

	
	/**
	 * Entry creation date as Unix timestamp (In seconds)
	 * @var int
	 * @readonly
	 */
	public $updatedAt;
		
	/**
	 * Enter description here...
	 *
	 * @var KalturaUiConfCreationMode $creationMode
	 */
	public $creationMode;
	

	private static $map_between_objects = array
	(
		"id" ,
		"creationMode" , 
	 	"objType" , "objTypeAsString" , "name" , "width" , "height" ,
		"htmlParams", "swfUrl" , "confFilePath" , "confFile" , "confFileFeatures" , "confVars" , "useCdn" , "tags" , "swfUrlVersion" , "createdAt" , "updatedAt" ,
        
	);

	public function getMapBetweenObjects ( )
	{
		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
	}

	public function fromUiConf ( uiConf  $uiConf )
	{
		parent::fromObject( $uiConf );
	}
	
	public function toUiConf () 
	{
		$uiConf = new uiConf();
		return parent::toObject( $uiConf )	;
	}
}
?>

File: kaltura/api_v3/generator/output/dotNet/KalturaClient/Types/KalturaUiConf.cs
using System;
using System.Xml;
using System.Collections.Generic;

namespace Kaltura
{
	public class KalturaUiConf : KalturaObjectBase
	{
		#region Properties
		public string Id = null;
		public string Name = null;
		public string Description = null;
		public int PartnerId = Int32.MinValue;
		public KalturaUiConfObjType ObjType = (KalturaUiConfObjType)Int32.MinValue;
		public string ObjTypeAsString = null;
		public int Width = Int32.MinValue;
		public int Height = Int32.MinValue;
		public string HtmlParams = null;
		public string SwfUrl = null;
		public string ConfFilePath = null;
		public string ConfFile = null;
		public string ConfFileFeatures = null;
		public string ConfVars = null;
		public bool? UseCdn = false;
		public string Tags = null;
		public string SwfUrlVersion = null;
		public int CreatedAt = Int32.MinValue;
		public int UpdatedAt = Int32.MinValue;
		public KalturaUiConfCreationMode CreationMode = (KalturaUiConfCreationMode)Int32.MinValue;
		#endregion

		#region CTor
		public KalturaUiConf()
		{
		}

		public KalturaUiConf(XmlElement node)
		{
			foreach (XmlElement propertyNode in node.ChildNodes)
			{
				string txt = propertyNode.InnerText;
				switch (propertyNode.Name)
				{
					case "id":
						this.Id = txt;
						continue;
					case "name":
						this.Name = txt;
						continue;
					case "description":
						this.Description = txt;
						continue;
					case "partnerId":
						this.PartnerId = ParseInt(txt);
						continue;
					case "objType":
						this.ObjType = (KalturaUiConfObjType)ParseEnum(typeof(KalturaUiConfObjType), txt);
						continue;
					case "objTypeAsString":
						this.ObjTypeAsString = txt;
						continue;
					case "width":
						this.Width = ParseInt(txt);
						continue;
					case "height":
						this.Height = ParseInt(txt);
						continue;
					case "htmlParams":
						this.HtmlParams = txt;
						continue;
					case "swfUrl":
						this.SwfUrl = txt;
						continue;
					case "confFilePath":
						this.ConfFilePath = txt;
						continue;
					case "confFile":
						this.ConfFile = txt;
						continue;
					case "confFileFeatures":
						this.ConfFileFeatures = txt;
						continue;
					case "confVars":
						this.ConfVars = txt;
						continue;
					case "useCdn":
						this.UseCdn = ParseBool(txt);
						continue;
					case "tags":
						this.Tags = txt;
						continue;
					case "swfUrlVersion":
						this.SwfUrlVersion = txt;
						continue;
					case "createdAt":
						this.CreatedAt = ParseInt(txt);
						continue;
					case "updatedAt":
						this.UpdatedAt = ParseInt(txt);
						continue;
					case "creationMode":
						this.CreationMode = (KalturaUiConfCreationMode)ParseEnum(typeof(KalturaUiConfCreationMode), txt);
						continue;
				}
			}
		}
		#endregion

		#region Methods
		public new KalturaParams ToParams()
		{
			KalturaParams kparams = base.ToParams();
			kparams.AddStringIfNotNull("id", this.Id);
			kparams.AddStringIfNotNull("name", this.Name);
			kparams.AddStringIfNotNull("description", this.Description);
			kparams.AddIntIfNotNull("partnerId", this.PartnerId);
			kparams.AddEnumIfNotNull("objType", this.ObjType);
			kparams.AddStringIfNotNull("objTypeAsString", this.ObjTypeAsString);
			kparams.AddIntIfNotNull("width", this.Width);
			kparams.AddIntIfNotNull("height", this.Height);
			kparams.AddStringIfNotNull("htmlParams", this.HtmlParams);
			kparams.AddStringIfNotNull("swfUrl", this.SwfUrl);
			kparams.AddStringIfNotNull("confFilePath", this.ConfFilePath);
			kparams.AddStringIfNotNull("confFile", this.ConfFile);
			kparams.AddStringIfNotNull("confFileFeatures", this.ConfFileFeatures);
			kparams.AddStringIfNotNull("confVars", this.ConfVars);
			kparams.AddBoolIfNotNull("useCdn", this.UseCdn);
			kparams.AddStringIfNotNull("tags", this.Tags);
			kparams.AddStringIfNotNull("swfUrlVersion", this.SwfUrlVersion);
			kparams.AddIntIfNotNull("createdAt", this.CreatedAt);
			kparams.AddIntIfNotNull("updatedAt", this.UpdatedAt);
			kparams.AddEnumIfNotNull("creationMode", this.CreationMode);
			return kparams;
		}
		#endregion
	}
}

